objects/obj_enemy_chasing/Step_0.gml

```diff
@@ -0,0 +1,97 @@
+/// @description Insert description here
+// You can write your code in this editor
+
+
+
+// Inherit the parent event
+event_inherited();
+
+at_edge = (!collision_point(x + sprite_width / 2, y + (sprite_height / 2), obj_ground, false, false))or (!collision_point(x - sprite_width / 2, y + (sprite_height / 2), obj_ground, false, false)); 
+at_wall = place_meeting(x + spd, y, obj_ground_parent);
+on_ground = instance_place(x,y+1,obj_ground);
+end_chasing = false;
+edge_side = 0; //left is 0,right is 1;
+
+if(at_edge and collision_point(x + sprite_width / 2, y + (sprite_height / 2), obj_ground, false, false)){
+ edge_side = 0;
+} 
+
+else if(at_edge and collision_point(x - sprite_width / 2, y + (sprite_height / 2), obj_ground, false, false)){
+ edge_side = 1;
+}
+
+
+
+
+var player_instance = instance_nearest(x, y, obj_player);
```

Comment:
Move this variable to the create method so we aren't performing this function every frame.
```suggestion
var player_instance = instance_nearest(x, y, obj_player);
```